pkg/config/template_test.go

```diff
@@ -0,0 +1,111 @@
+package config
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	templateapi "github.com/openshift/api/template/v1"
+	templatescheme "github.com/openshift/client-go/template/clientset/versioned/scheme"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	coretesting "k8s.io/client-go/testing"
+)
+
+const templatesPath = "../../test/pj-rehearse-integration/master/ci-operator/templates"
+
+func TestGetTemplates(t *testing.T) {
+	expectCiTemplates := getBaseCiTemplates(t)
+	if templates, err := getTemplates(templatesPath); err != nil {
+		t.Fatal(err)
+	} else if !equality.Semantic.DeepEqual(templates, expectCiTemplates) {
+		t.Fatalf("Diff found %s", diff.ObjectReflectDiff(expectCiTemplates, templates))
+	}
+}
+
+var (
+	expectedCmName   = "rehearse-1234-test-template"
+	expectedCmLabels = map[string]string{rehearseLabel: "rehearse-temporary-cm-1234"}
+)
+
+func TestCreateCleanupCMTemplates(t *testing.T) {
+	req, err := labels.NewRequirement(rehearseLabel, selection.Equals, []string{"rehearse-temporary-cm-1234"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	selector := labels.NewSelector().Add(*req)
+
+	expectedListRestricitons := coretesting.ListRestrictions{
+		Labels: selector,
+	}
+
+	cs := &fake.FakeCoreV1{Fake: &coretesting.Fake{}}
+	cs.Fake.AddReactor("create", "configmaps", func(action coretesting.Action) (handled bool, ret runtime.Object, err error) {
+		createAction := action.(coretesting.CreateAction)
+		cm := createAction.GetObject().(*v1.ConfigMap)
+
+		if cm.ObjectMeta.Name != expectedCmName {
+			t.Fatalf("Configmap name:\nExpected: %s\nFound: %s", expectedCmName, cm.ObjectMeta.Name)
+		}
+
+		if !reflect.DeepEqual(cm.ObjectMeta.Labels, expectedCmLabels) {
+			t.Fatalf("Configmap labels\nExpected: %#v\nFound: %#v", expectedCmLabels, cm.ObjectMeta.Labels)
+		}
+
+		return true, nil, nil
+	})
+	cs.Fake.AddReactor("*", "configmaps", func(action coretesting.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
you bind this to all actions, not just delete, so it will panic if someone changes the code under test

Reply:
I wasn't able to find the correct verb for `DeleteCollection`. I will take a further look at it.

Reply:
@deads2k do you know?

Reply:
"deletecollection" verb doesn't work. Bug in upstream? I will investigate

Reply:
Ok I found it. `delete-collection` :)